File: src/net/Defracted/AdminUtils/commands/God.java
package net.Defracted.AdminUtils.commands;

import net.Defracted.AdminUtils.Main;
import net.Defracted.AdminUtils.util.Formatters;
import org.bukkit.Bukkit;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;


public class God implements CommandExecutor {
    static Main plugin;

    public God(Main instance) {
        plugin = instance;
    }

    public boolean onCommand(CommandSender sender, Command cmd, String label, String[] args) {
        if (cmd.getName().equalsIgnoreCase("god")) {
            if (!(sender instanceof Player)) {
                return true;
            }

            Player p = (Player) sender;
            Player target;

            if (args.length <= 0) {
                target = p;
            } else {
                target = Bukkit.getPlayerExact(args[0]);

                if (target == null) {
                    p.sendMessage(Formatters.chat("&eИгрок &c" + args[0] + " &eне найден."));
                    return true;
                }
            }

            if (!(plugin.playersInGodMode.contains(target.getUniqueId()))) {
                plugin.playersInGodMode.add(target.getUniqueId());

                if (target != p) {
                    p.sendMessage(Formatters.chat("&eВы &aвключили неуязвимсть для игрока &d" + target.getName() + "&e."));
                }

                target.sendMessage(Formatters.chat("&eВы &aвключили &eнеуязвимость."));

            } else {
                plugin.playersInGodMode.remove(target.getUniqueId());

                if (target != p) {
                    p.sendMessage(Formatters.chat("&eВы &cвыключили неуязвимсть для игрока &d" + target.getName() + "&e."));
                }

                target.sendMessage(Formatters.chat("&eВы &cвыключили &eнеуязвимость."));
            }

            return true;
        }

        return false;
    }
}

File: src/main/kotlin/net/defracted/adminutils/commands/God.kt
package net.defracted.adminutils.commands

import net.defracted.adminutils.Main
import net.defracted.adminutils.util.Formatters.chat

import org.bukkit.Bukkit
import org.bukkit.Sound
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player


class God(private val plugin: Main) : CommandExecutor {
    override fun onCommand(sender: CommandSender, cmd: Command, label: String, args: Array<out String>): Boolean {
        if (cmd.name.equals("god", ignoreCase = true)) {
            if (sender !is Player) {
                return true
            }
            val p: Player = sender
            val target: Player?
            if (args.isEmpty()) {
                target = p
            } else {
                if (!p.hasPermission("adminutils.god.others")) {
                    p.sendMessage(chat("&cВы не можете управлять неуязвимостью других игроков!"))
                    p.playSound(p.location, Sound.ENTITY_ENDERMAN_TELEPORT, 1f, 0f)
                    return true
                }

                target = Bukkit.getPlayerExact(args[0])
                if (target == null) {
                    p.sendMessage(chat("&cИгрок &f" + args[0] + " &cне найден."))
                    return true
                }
            }

            if (!plugin.godModePlayers.contains(target.uniqueId)) {
                plugin.godModePlayers.add(target.uniqueId)
                if (target !== p) {
                    p.sendMessage(chat("&cВы &aвключили неуязвимсть для игрока &f" + target.name + "&c."))
                }
                target.sendMessage(chat("&bВы &aвключили &bнеуязвимость."))
            } else {
                plugin.godModePlayers.remove(target.uniqueId)
                if (target !== p) {
                    p.sendMessage(chat("&cВы &4выключили неуязвимсть для игрока &f" + target.name + "&c."))
                }
                target.sendMessage(chat("&bВы &cвыключили &bнеуязвимость."))
            }
            return true
        }

        return false
    }
}